### PR TITLE
Feature/azure docstore hotfix

### PR DESCRIPTION
**Pull request by Falven**
# Description

Fixes aget_ref_doc_info which was using the wrong column and not properly checking for empty results.

Fixes # (issue)

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ ] Yes
- [x] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [x] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] I stared at the code and made sure it makes sense
- [x] Ran it on my own infrastructure

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] New and existing unit tests pass locally with my changes
- [X] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-core/llama_index/core/storage/docstore/keyval_docstore.py [hunk 1]
```diff
@@ -206,7 +206,7 @@ def _prepare_kv_pairs(
 
     def add_documents(
         self,
-        nodes: Sequence[BaseNode],
+        docs: Sequence[BaseNode],
```
Comment (logan-markewich) [inline llama-index-core/llama_index/core/storage/docstore/keyval_docstore.py:209]: why rename this? Technically it is nodes 😅

**Review by logan-markewich — COMMENTED** [review]
I agree with the fixes to azure, but don't really agree with the changes to core

File: llama-index-core/llama_index/core/storage/docstore/keyval_docstore.py (hunk 1 above)
Comment (Falven) [reply llama-index-core/llama_index/core/storage/docstore/keyval_docstore.py:209]: > why rename this? Technically it is nodes 😅

I agree with you 100%. The problem I encountered was wherever the parameter is typed as the base class, if we provide what the typing information says: `docs: Sequence[BaseNode],` you will get a nasty error where your kwarg was named `docs`, but you provided `nodes`.

```py
@abstractmethod
    def add_documents(
        self,
        docs: Sequence[BaseNode],
        allow_update: bool = True,
        batch_size: int = DEFAULT_BATCH_SIZE,
        store_text: bool = True,
    ) -> None:
        ...
```

Comment (logan-markewich) [reply llama-index-core/llama_index/core/storage/docstore/keyval_docstore.py:209]: right, but its a positional argument, so it doesn't need to be provided as a kwarg?

Comment (logan-markewich) [reply llama-index-core/llama_index/core/storage/docstore/keyval_docstore.py:209]: `add_documents(nodes, ...)` should work fine no?

Comment (Falven) [reply llama-index-core/llama_index/core/storage/docstore/keyval_docstore.py:209]: Correct, but if somebody does try to provide it as a kwarg they will get an error that can be difficult to figure out. I'm happy to change the base class to nodes and try to find everywhere else that docs is used.

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**